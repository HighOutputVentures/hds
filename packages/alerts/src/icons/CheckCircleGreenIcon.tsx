import { Icon, IconProps } from '@chakra-ui/react';
import React, { FC } from 'react';

const CheckCircleGreenIcon: FC<Omit<IconProps, 'children' | 'css'>> = (
  props
) => (
  <Icon width="20px" height="20px" viewBox="0 0 20 20" fill="none" {...props}>
    <path
      d="M6.25008 9.99996L8.75008 12.5L13.7501 7.49996M18.3334 9.99996C18.3334 14.6023 14.6025 18.3333 10.0001 18.3333C5.39771 18.3333 1.66675 14.6023 1.66675 9.99996C1.66675 5.39759 5.39771 1.66663 10.0001 1.66663C14.6025 1.66663 18.3334 5.39759 18.3334 9.99996Z"
      stroke="#139E19"
      stroke-width="1.66667"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </Icon>
);

export default CheckCircleGreenIcon;
