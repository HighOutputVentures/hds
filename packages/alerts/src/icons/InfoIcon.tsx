import { Icon, IconProps } from '@chakra-ui/react';
import React, { FC } from 'react';

const InfoIcon: FC<Omit<IconProps, 'children' | 'css'>> = (props) => (
  <Icon width="20px" height="20px" viewBox="0 0 20 20" fill="none" {...props}>
    <path
      d="M10.0001 13.3334V10M10.0001 6.66669H10.0084M18.3334 10C18.3334 14.6024 14.6025 18.3334 10.0001 18.3334C5.39771 18.3334 1.66675 14.6024 1.66675 10C1.66675 5.39765 5.39771 1.66669 10.0001 1.66669C14.6025 1.66669 18.3334 5.39765 18.3334 10Z"
      stroke="#4A3880"
      stroke-width="1.66667"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </Icon>
);

export default InfoIcon;
