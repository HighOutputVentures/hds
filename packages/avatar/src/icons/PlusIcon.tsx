import * as React from "react";

export default function PlusIcon(props: React.ComponentProps<"svg">) {
  return (
    <svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
      <path
        d="M7.99992 3.33331V12.6666M3.33325 7.99998H12.6666"
        stroke="currentColor"
        strokeWidth="1.33333"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
}
