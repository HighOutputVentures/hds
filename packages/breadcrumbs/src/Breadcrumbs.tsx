import {
  Box,
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  Flex,
  HStack,
  Text,
} from '@chakra-ui/react';
import React from 'react';
import { useTransformBreadCrumbInfo } from './hooks';

export interface IBreadCrumbLinks {
  typedef: Array<{ name: string; link: string } | null | undefined>;
}
export interface BreadcrumbProps {
  maxLinkControls: 6 | 4 | 2;
  breadCrumbLinks: IBreadCrumbLinks['typedef'];
  separator:
    | string
    | React.ReactElement<any, string | React.JSXElementConstructor<any>>;
  icon: any;
  type: 'color-in-text' | 'color-on-bg' | 'default';
}
const Breadcrumbs = (props: BreadcrumbProps) => {
  const { maxLinkControls = 6, separator, breadCrumbLinks, icon, type } = props;
  const [start, setStart] = React.useState(0);
  const { transformedBreadCrumbData } = useTransformBreadCrumbInfo({
    start,
    maxLinkControls,
    originalBreadCrumbDaTa: breadCrumbLinks,
  });
  const [active, setActive] = React.useState('#');

  const textColor = (link: string) => {
    if (active === link && type === 'color-in-text') return '#4A3880';
    if (active === link && type === 'color-on-bg') return '#4A3880';
    if (active === link && type === 'default') return 'neutrals.900';
    else return 'neutrals.600';
  };

  return (
    <HStack
      height={'32.67px'}
      align="center"
      bgColor={'#FCFCFC'}
      padding={'4px'}
      borderRadius={'8px'}
      color={'neutrals.600'}
      data-testid="breadcrumb"
    >
      <Flex justify="end" align={'end'} gap="14px">
        <Box position={'relative'} top={'-2px'}>
          {icon}
        </Box>
        <Box> {separator}</Box>
      </Flex>
      <Breadcrumb spacing="14px" separator={separator}>
        {transformedBreadCrumbData!.map((d, idx) => {
          if (d === undefined) {
            return null;
          }
          return (
            <BreadcrumbItem textDecoration={'none'} key={idx}>
              <BreadcrumbLink
                href={d?.link ?? undefined}
                onClick={() =>
                  d === null
                    ? setStart(idx === 0 ? start - 1 : start + 1)
                    : setActive(d.link)
                }
                textDecorationLine={'none'}
                color={textColor(d?.link!)}
                bgColor={
                  active === d?.link && type === 'color-on-bg'
                    ? '#EDE8FC'
                    : 'none'
                }
                padding={
                  active === d?.link && type === 'color-on-bg'
                    ? '4px 8px'
                    : 'none'
                }
                borderRadius={'6px'}
              >
                <Text textDecoration={'none'}>
                  {d === null ? '...' : d?.name}
                </Text>
              </BreadcrumbLink>
            </BreadcrumbItem>
          );
        })}
      </Breadcrumb>
    </HStack>
  );
};

export default Breadcrumbs;
