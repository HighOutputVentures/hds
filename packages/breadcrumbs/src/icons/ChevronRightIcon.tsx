import * as React from 'react';

export default function ChevronRightIcon(props: React.ComponentProps<'svg'>) {
  return (
    <svg
      width="17"
      height="17"
      viewBox="0 0 17 17"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M6.66675 12.3333L10.6667 8.33331L6.66675 4.33331"
        stroke="currentColor"
        strokeWidth="1.33333"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
}
