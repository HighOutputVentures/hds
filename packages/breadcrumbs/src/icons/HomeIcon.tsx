import * as React from 'react';

export default function HomeIcon(props: React.ComponentProps<'svg'>) {
  return (
    <svg
      width="17"
      height="17"
      viewBox="0 0 17 17"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M5.55547 11.8056H11.111M7.6511 1.91946L2.94116 5.58275C2.62632 5.82762 2.4689 5.95006 2.35549 6.10339C2.25503 6.23922 2.1802 6.39223 2.13466 6.55492C2.08325 6.73858 2.08325 6.938 2.08325 7.33686V12.3611C2.08325 13.139 2.08325 13.5279 2.23463 13.825C2.36779 14.0863 2.58026 14.2988 2.8416 14.432C3.1387 14.5833 3.52762 14.5833 4.30547 14.5833H12.361C13.1389 14.5833 13.5278 14.5833 13.8249 14.432C14.0862 14.2988 14.2987 14.0863 14.4319 13.825C14.5833 13.5279 14.5833 13.139 14.5833 12.3611V7.33686C14.5833 6.938 14.5833 6.73858 14.5318 6.55492C14.4863 6.39223 14.4115 6.23922 14.311 6.10339C14.1976 5.95006 14.0402 5.82762 13.7253 5.58275L9.01541 1.91946C8.77143 1.72971 8.64944 1.63483 8.51474 1.59835C8.39589 1.56617 8.27062 1.56617 8.15176 1.59835C8.01706 1.63483 7.89507 1.7297 7.6511 1.91946Z"
        stroke="currentColor"
        strokeWidth="1.66667"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
}
