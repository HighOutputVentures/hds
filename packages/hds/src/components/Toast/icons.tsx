import * as React from 'react';

export function CheckCircleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        clipRule="evenodd"
        fillRule="evenodd"
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zm13.36-1.814a.75.75 0 10-1.22-.872l-3.236 4.53L9.53 12.22a.75.75 0 00-1.06 1.06l2.25 2.25a.75.75 0 001.14-.094l3.75-5.25z"
      />
    </svg>
  );
}

export function ExclamationCircleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        clipRule="evenodd"
        fillRule="evenodd"
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z"
      />
    </svg>
  );
}

export function XMarkIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        clipRule="evenodd"
        fillRule="evenodd"
        d="M5.47 5.47a.75.75 0 011.06 0L12 10.94l5.47-5.47a.75.75 0 111.06 1.06L13.06 12l5.47 5.47a.75.75 0 11-1.06 1.06L12 13.06l-5.47 5.47a.75.75 0 01-1.06-1.06L10.94 12 5.47 6.53a.75.75 0 010-1.06z"
      />
    </svg>
  );
}

export function InformationCircleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zm8.706-1.442c1.146-.573 2.437.463 2.126 1.706l-.709 2.836.042-.02a.75.75 0 01.67 1.34l-.04.022c-1.147.573-2.438-.463-2.127-1.706l.71-2.836-.042.02a.75.75 0 11-.671-1.34l.041-.022zM12 9a.75.75 0 100-1.5.75.75 0 000 1.5z"
        fillRule="evenodd"
        clipRule="evenodd"
      />
    </svg>
  );
}

export function ExclamationTriangleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        d="M9.401 3.003c1.155-2 4.043-2 5.197 0l7.355 12.748c1.154 2-.29 4.5-2.599 4.5H4.645c-2.309 0-3.752-2.5-2.598-4.5L9.4 3.003zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z"
        fillRule="evenodd"
        clipRule="evenodd"
      />
    </svg>
  );
}

export function SpinnerIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      {...props}
    >
      <path
        fill="currentColor"
        d="M12,1A11,11,0,1,0,23,12,11,11,0,0,0,12,1Zm0,19a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
        opacity=".25"
      ></path>
      <path
        fill="currentColor"
        d="M12,4a8,8,0,0,1,7.89,6.7A1.53,1.53,0,0,0,21.38,12h0a1.5,1.5,0,0,0,1.48-1.75,11,11,0,0,0-21.72,0A1.5,1.5,0,0,0,2.62,12h0a1.53,1.53,0,0,0,1.49-1.3A8,8,0,0,1,12,4Z"
      >
        <animateTransform
          attributeName="transform"
          dur="0.75s"
          repeatCount="indefinite"
          type="rotate"
          values="0 12 12;360 12 12"
        ></animateTransform>
      </path>
    </svg>
  );
}

export const Number123Icon = React.forwardRef<
  SVGSVGElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return (
    <svg
      ref={ref}
      xmlns="http://www.w3.org/2000/svg"
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      {...props}
    >
      <path
        fill="none"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        d="m3 10l2-2v8m4-8h3a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1h-2a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h3m4-8h2.5A1.5 1.5 0 0 1 21 9.5v1a1.5 1.5 0 0 1-1.5 1.5H18h1.5a1.5 1.5 0 0 1 1.5 1.5v1a1.5 1.5 0 0 1-1.5 1.5H17"
      ></path>
    </svg>
  );
});
