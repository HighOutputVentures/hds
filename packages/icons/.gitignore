*.log
.DS_Store
node_modules
.cache
!.vscode
storybook-static
.vercel
