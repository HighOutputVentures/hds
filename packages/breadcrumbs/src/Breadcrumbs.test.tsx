import { AttachmentIcon, ChevronRightIcon } from '@chakra-ui/icons';
import { ThemeProvider } from '@highoutput/hds';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Breadcrumbs from './Breadcrumbs';

describe('Breadcrumbs Component', () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <Breadcrumbs
          separator={<ChevronRightIcon color="#D6D6D6" />}
          maxLinkControls={2}
          icon={<AttachmentIcon />}
          breadCrumbLinks={[
            { link: '#hello', name: 'Home' },
            { link: '#Settings', name: 'Settings' },
            { link: '#Faqs', name: 'Faqs' },
            { link: '#Logout', name: 'Logout' },
            { link: '#New', name: 'New' },
            { link: '#Create', name: 'Create' },
            { link: '#Contact', name: 'Contact' },
            { link: '#Read', name: 'Read' },
            { link: '#Save', name: 'Save' },
            { link: '#Copy', name: 'Copy' },
          ]}
        />
      </ThemeProvider>
    );
  });

  it('should render breadcrumb', async () => {
    const breadcrumb = screen.getByTestId('hds.breadcrumb');
    expect(breadcrumb).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const component = renderer.create(
      <ThemeProvider>
        <Breadcrumbs
          separator={<ChevronRightIcon color="#D6D6D6" />}
          maxLinkControls={2}
          icon={<AttachmentIcon />}
          breadCrumbLinks={[
            { link: '#hello', name: 'Home' },
            { link: '#Settings', name: 'Settings' },
            { link: '#Faqs', name: 'Faqs' },
            { link: '#Logout', name: 'Logout' },
            { link: '#New', name: 'New' },
            { link: '#Create', name: 'Create' },
            { link: '#Contact', name: 'Contact' },
            { link: '#Read', name: 'Read' },
            { link: '#Save', name: 'Save' },
            { link: '#Copy', name: 'Copy' },
          ]}
        />
      </ThemeProvider>
    );

    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
    component.unmount();
  });
});
