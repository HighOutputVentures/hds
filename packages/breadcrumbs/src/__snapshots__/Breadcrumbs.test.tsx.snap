// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component Should match snapshot 1`] = `
Array [
  <div
    className="chakra-stack css-1xcp7r9"
    data-testid="breadcrumb"
  >
    <div
      className="css-1ktsxmj"
    >
      <div
        className="css-1czdu59"
      >
        <svg
          className="chakra-icon css-onkibi"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M21.843,3.455a6.961,6.961,0,0,0-9.846,0L1.619,13.832a5.128,5.128,0,0,0,7.252,7.252L17.3,12.653A3.293,3.293,0,1,0,12.646,8L7.457,13.184A1,1,0,1,0,8.871,14.6L14.06,9.409a1.294,1.294,0,0,1,1.829,1.83L7.457,19.67a3.128,3.128,0,0,1-4.424-4.424L13.411,4.869a4.962,4.962,0,1,1,7.018,7.018L12.646,19.67a1,1,0,1,0,1.414,1.414L21.843,13.3a6.96,6.96,0,0,0,0-9.846Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="css-0"
      >
         
        <svg
          className="chakra-icon css-1d2sseo"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <nav
      aria-label="breadcrumb"
      className="chakra-breadcrumb css-0"
    >
      <ol
        className="chakra-breadcrumb__list css-70qvj9"
      >
        <li
          className="chakra-breadcrumb__list-item css-aou3wz"
        >
          <a
            className="chakra-breadcrumb__link css-9phpk7"
            href="#hello"
            onClick={[Function]}
          >
            <p
              className="chakra-text css-ytumd6"
            >
              Home
            </p>
          </a>
          <span
            className="css-1xgwh2j"
            role="presentation"
          >
            <svg
              className="chakra-icon css-1d2sseo"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                fill="currentColor"
              />
            </svg>
          </span>
        </li>
        <li
          className="chakra-breadcrumb__list-item css-aou3wz"
        >
          <a
            className="chakra-breadcrumb__link css-9phpk7"
            onClick={[Function]}
          >
            <p
              className="chakra-text css-ytumd6"
            >
              ...
            </p>
          </a>
          <span
            className="css-1xgwh2j"
            role="presentation"
          >
            <svg
              className="chakra-icon css-1d2sseo"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                fill="currentColor"
              />
            </svg>
          </span>
        </li>
        <li
          className="chakra-breadcrumb__list-item css-aou3wz"
        >
          <a
            className="chakra-breadcrumb__link css-9phpk7"
            href="#Copy"
            onClick={[Function]}
          >
            <p
              className="chakra-text css-ytumd6"
            >
              Copy
            </p>
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  <span />,
]
`;
